students/Dkuchan/ClassProject/PythonClassProject.py

```diff
@@ -0,0 +1,625 @@
+        self.kpa = []
+        self.pilot = []
+        self.alliances = []
+        self.fieldpos = []
+        self.autogears = []
+        self.autogearsinpos = []
+        self.autokpa = []
+        self.autokpainpos = []
+        self.climb = []
+        self.matchwins = []
+
+   # def __repr__(self):
+    #    return '{}, {}: {}'.format(self.team, self.matchnumber, self.win)
+
+    def __str__(self):
+        return 'Team Number: {}, Team Name: {}'.format(self.number, self.name)
+
+    def fillmatches(self):
+        for i in range(0, len(matchlist)):   # I wanted to do this in a list
+            if int(matchlist[i].team) == int(self.number):  # did your team play in this match
+                self.matchesplayed.append(matchlist[i].matchnumber)  # if so append the list comprehensions but I is a match type variable
+
+    def fillmatchwins(self):
+        for i in range(0, len(matchlist)):   # I wanted to do this in a list
+            if matchlist[i].team == self.number:  # did your team play in this match
+                if int(matchlist[i].win) == 1:
+                    self.matchwins.append(int(matchlist[i].win))
+    @property
+    def nummatches(self):
+        return len(self.matchesplayed)
+
+    @property
+    def averagegears(self):
+        return (sum(self.gearsdelivered) / len(self.gearsdelivered))
+
+    @property
+    def maxgearsdelivered(self):
+        return max(self.gearsdelivered)
+
+    @property
+    def mingearsdelivered(self):
+        return min(self.gearsdelivered)
+
+    @property
+    def averagekpa(self):
+        return sum(self.kpa) / len(self.kpa)
+
+    @property
+    def maxkpa(self):
+        return max(self.kpa)
+
+    @property
+    def minkpa(self):
+        return min(self.kpa)
+
+    @property
+    def climbpercent(self):
+        return (len(self.climb) / len(self.matchesplayed) * 100)
+
+#    @property
+ #   def autogearsinpos(self):
+        # This creates a dictionsary with postion as Key and Autogears scored as value DOES NOT WORK YET
+  #      for i in range(1, 3):
+   #         for j in self.matchesplayed:
+    #            if self.fieldpos[j] == i:
+     #               self.autogearsinpos[i] = self.autogears[j]
+      #  return self.autogearsinpos
+
+    #@property
+    #def autokpainpos(self):
+        # This creates a dictionsary with postion as Key and Autogears scored as value
+     #   for i in range(1, 3):
+      #      for j in self.matchesplayed:
+       #         if self.fieldpos[j] == i:
+        #            autokpainpos[i] = self.autokpa[j]
+        #return autokpainpos
+
+    
+
+
+                    #self.matchwins = {matchlist[i].matchnumber: matchlist[i].win for i in matchlist if matchlist[i].teamnumber == self.number}        
+       # self.matchesplayed = [int(matchlist[i].matchnumber) for i in matchlist if int(matchlist[i].teamnumber) == self.number]
+
+
+    def fillgearsdelivered(self):
+        for i in range(0, len(matchlist)):   #I wanted to do this in a list 
```

Comment:
looking at these loops, it seems maybe keeping the matchlist up to date when each match is created would make more sense. Maybe have an object that represents the tournament?